### ISSUE TITLE
Add discount descriptions to Kane's invoice

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1012]
_Imported from trac ticket 1012.
Created by **kbower**
Opened in trac: Aug 23, 2011, 1:22:56 PM
Last modified in trac: Jul 23, 2012, 3:30:20 PM_
***
_by kbower:_
Was also requested as part of #857
----
Need the ability to add all discount descriptions to our invoices. 

***
* kbower attached [Error Popup.JPG](https://trac.retailarchitects.com/trac/attachment/ticket/1012/Error Popup.JPG) on Jul 23, 2012, 12:59:07 PM

**Comment by kbower** [comment]
_**rneville** trac comment on Jun 14, 2012, 3:05:10 PM:_

Calvin, may already be done and just needs the ticket cleaned up

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 18, 2012, 8:26:33 AM:_

See Sameer

**Comment by kbower** [comment]
_**criley** trac comment on Jul 20, 2012, 3:32:46 PM:_

(Changeset [[2898]](../commit/01b48f53269a0d4310c5e8d3819b36d7e671d0a5))
*    Fixes #1012 (Add discount descriptions to Kane's invoice)
*    Retouches [#1698](1698) (ITM.GRADE printed twice)

**Comment by kbower** [comment]
_**criley** trac status change on Jul 20, 2012, 3:32:46 PM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac status change on Jul 23, 2012, 12:57:47 PM:_ **reopened**

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 12:59:03 PM:_

see [r2905](../commit/ce497e3425a72d2e0229429c1c7a96cc645504f5)

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 12:59:07 PM:_

Kent,

The error in this attachment came up right at the end of an order while trying tro Save. It also appeared on another PC going from the delivery date to Make a Payment. This was after the server re-boot.

Also had speed issues with SKU searches as well ad when we did a price override (approx 45 seconds for both)

Tim

[attachment:"Error Popup.JPG"]

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 3:12:22 PM:_

```
  File "/home/rarch/trunk/src/appserver/pylotengine/templates/../../reports/kan_receipt.html", line 248, in <Expression '?'>
    <py:otherwise py:with="desc = line.product.displaydescription.replace('(' + line.product.grade + ')', line.product.grade)">
TypeError: cannot concatenate 'str' and 'NoneType' objects
```
[[Image(Error Popup.JPG)]]

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 3:30:20 PM:_

(Changeset [[2908]](../commit/37b339ce9e914aec3b0583ccb77312a9cd95d6a6))
 * fix #1012, move kane's specific code out of objects.py

**Comment by kbower** [comment]
_**kbower** trac status change on Jul 23, 2012, 3:30:20 PM:_ **closed**
